qiita_db/metadata_template.py

```diff
@@ -1324,6 +1355,95 @@ def study_id(self):
         """
         return self._id
 
+    def extend(self, md_template):
+        """Adds the given sample template to the current one
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by samples Ids
+        """
+        conn_handler = SQLConnectionHandler()
+        queue_name = "EXTEND_SAMPLE_TEMPLATE_%d" % self.id
+        conn_handler.create_queue(queue_name)
+
+        md_template = self._clean_validate_template(md_template, self.study_id,
+                                                    conn_handler)
+
+        # Get some useful information from the metadata template
+        sample_ids = md_template.index.tolist()
```

Comment:
This is a copy paste of what was originally there for validation, so it's probably fine. If you look a few lines down it's used for the intersection with curr_samples, so it being a set or list shouldn't make a difference on performance.